scripts/datasets/DET.py

```diff
@@ -0,0 +1,276 @@
+    if not isdir(crop_path): makedirs(crop_path)
+    VID_base_path = join(args.download_dir, './ILSVRC')
+    ann_base_path = join(VID_base_path, 'Annotations/DET/train/')
+    sub_sets = ('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i')
+    for sub_set in sub_sets:
+        sub_set_base_path = join(ann_base_path, sub_set)
+        if 'a' == sub_set:
+            xmls = sorted(glob.glob(join(sub_set_base_path, '*', '*.xml')))
+        else:
+            xmls = sorted(glob.glob(join(sub_set_base_path, '*.xml')))
+        n_imgs = len(xmls)
+        sub_set_crop_path = join(crop_path, sub_set)
+        with futures.ProcessPoolExecutor(max_workers=args.num_threads) as executor:
+            fs = [executor.submit(crop_xml, args, xml, sub_set_crop_path, args.instance_size) for xml in xmls]
+            for i, f in enumerate(futures.as_completed(fs)):
+                printProgress(i, n_imgs, prefix=sub_set, suffix='Done ', barLength=80)
+
+def gen_json(args):
+    """Format XML and transform json.
+       generate train and val json, prepare for tracking dataloader"""
+    js = {}
+    VID_base_path = join(args.download_dir, './ILSVRC')
+    ann_base_path = join(VID_base_path, 'Annotations/DET/train/')
+    sub_sets = ('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i')
+    for sub_set in sub_sets:
+        sub_set_base_path = join(ann_base_path, sub_set)
+        if 'a' == sub_set:
+            xmls = sorted(glob.glob(join(sub_set_base_path, '*', '*.xml')))
+        else:
+            xmls = sorted(glob.glob(join(sub_set_base_path, '*.xml')))
+        n_imgs = len(xmls)
+        for f, xml in enumerate(xmls):
+            print('subset: {} frame id: {:08d} / {:08d}'.format(sub_set, f, n_imgs))
+            xmltree = ET.parse(xml)
+            objects = xmltree.findall('object')
+
+            video = join(sub_set, xml.split('/')[-1].split('.')[0])
+
+            for id, object_iter in enumerate(objects):
+                bndbox = object_iter.find('bndbox')
+                bbox = [int(bndbox.find('xmin').text), int(bndbox.find('ymin').text),
+                        int(bndbox.find('xmax').text), int(bndbox.find('ymax').text)]
+                frame = '%06d' % (0)
+                obj = '%02d' % (id)
+                if video not in js:
+                    js[video] = {}
+                if obj not in js[video]:
+                    js[video][obj] = {}
+                js[video][obj][frame] = bbox
+
+    train = {k:v for (k, v) in js.items() if 'i/' not in k}
+    val = {k:v for (k, v) in js.items() if 'i/' in k}
+    json.dump(train, open(join(args.download_dir, 'train.json'), 'w'),
+              indent=4, sort_keys=True)
+    json.dump(val, open(join(args.download_dir, 'val.json'), 'w'),
+              indent=4, sort_keys=True)
+
+def symlink(args):
+    """Soft connection"""
+    chdir(args.download_dir)
+    system('ln -sfb $PWD/ILSVRC/Annotations/DET/train/ILSVRC2013_train ILSVRC/Annotations/DET/train/a')
```

Comment:
use os.symlink instead for sake of windows users

Reply:
Good to know, thank you.